wagtail/models/__init__.py

```diff
@@ -2227,6 +2227,10 @@ def get_indexed_instance(self):
         except self.specific_class.DoesNotExist:
             return None
 
+    def get_default_privacy_setting(self, request: HttpRequest):
+        """Set the default privacy setting for a page."""
+        return {"type": "public"}
```

Comment:
Shouldn’t this have logic to account for `private_page_options`? Avoid defaulting to `public` if it’s not in the list of allowed options.

Reply:
`private_page_options` is a list of privacy options in addition to `public`. This is because prior to adding this default privacy option there was no way to stop a page from being public (only to prevent it from being changed after it was created).

After `setting the default privacy option` gets added it would make sense to add to the list of privacy options `public` and return a default privacy setting based on that list. This would however be a backwards incompatible change because any model that had for example `private_page_options = ['password']` would allow `public` and `password` privacy options but after would only have 'password' as an option.
Being that the `private_page_options` was only added in 6.2 this breaking change shouldn't effect most sites. 
I am open to alternative options for disabling public. Maybe it should be a flag `disable_public_page`?

It would also need a design decision as to how to determining the default privacy option if `public` can be disabled? (should it be the first item in the private page options or should it check first public, then login, then groups, then shared password?)

In addition if the default setting ends up being shared password what would the default password be?